docs/guide/module-configuration-patterns.md

```diff
@@ -0,0 +1,332 @@
+
+The same customization patterns apply to all Apostrophe modules, including:
+- Core modules bundled with ApostropheCMS
+- Official packages from the Apostrophe team (like `@apostrophecms/blog`)
+- Third-party modules from other developers
+- Your own custom modules installed via npm
+
+When you install any module through npm, you should **never modify the files in `node_modules`**. Instead, use project-level improvements and extensions just as you would with core modules.
+
+### Example: Improving a Third-Party Module
+
+Say you've installed a module called `organization-chart` (a hypothetical Apostrophe module for displaying org charts):
+
+### Improving the Module
+
+You might want to improve the base module to add functionality that will apply to all uses of it:
+
+<AposCodeBlock>
+
+```javascript
+export default {
+  // No "extend" property - this improves the original module
+  methods(self) {
+    return {
+      // Add a new utility method to the original module
+      sortEmployeesByDepartment(employees) {
+        return employees.sort((a, b) => a.department.localeCompare(b.department));
+      }
+    }
+  },
+  handlers(self) {
+    return {
+      // Add an event handler to the original module
+      beforeSave: {
+        validateReportingStructure(req, piece) {
+          // Custom validation logic
+        }
+      }
+    }
+  }
+}
+```
+  <template v-slot:caption>
+    modules/organization-chart/index.js
+  </template>
+</AposCodeBlock>
+
+### Extending the Module
+
+At the same time, you could extend the module to create specialized versions of it:
+
+<AposCodeBlock>
+
+```javascript
+export default {
+  extend: 'organization-chart',
+  options: {
+    label: 'Executive Team Chart',
+    pluralLabel: 'Executive Team Charts'
+  },
+  fields: {
+    add: {
+      fiscalYear: {
+        type: 'select',
+        label: 'Fiscal Year',
+        choices: [
+          { label: '2023', value: '2023' },
+          { label: '2024', value: '2024' },
+          { label: '2025', value: '2025' }
+        ]
+      }
+    }
+  }
+}
+```
+  <template v-slot:caption>
+    modules/executive-team-chart/index.js
+  </template>
+</AposCodeBlock>
+
+### How This Works Together
+
+When Apostrophe initializes your project:
+
+1. It applies your improvements to the `organization-chart` module first
+2. Then it creates the new `executive-team-chart` module type that extends the (now improved) `organization-chart` module
+3. The `executive-team-chart` module inherits all the functionality from the improved base module, plus gets its own specialized fields and behavior
+
+This powerful pattern lets you:
+1. Add global improvements that affect all instances of a module
+2. Create specialized versions of modules for specific purposes
+3. Maintain clean separation of concerns in your code
+
+It's particularly useful when you want to both enhance a module's core functionality and create specialized variants of it for different use cases in your project.
```

Comment:
Since you've taken things this far, you could add a section covering how to ship an "improvement" as an npm module, where the name cannot be the same as the module being improved, and stressing that this is the **one and only situation** where `improve` can or should be used to indicate the name of the module you are improving.